pkg/certificate/providers/config.go

```diff
@@ -154,6 +154,45 @@ func (c *Config) GetCertificateManager() (certificate.Manager, debugger.Certific
 	}
 }
 
+// SynchronizeCertificate is a helper function that ensures creation and synchronization of a certificate
+// using Kubernetes Secrets backend and API atomicity.
+func SynchronizeCertificate(ns string, secretName string, cert certificate.Certificater, kubeClient kubernetes.Interface) (certificate.Certificater, error) {
```

Comment:
```suggestion
func GetCertificateFromSecret(ns string, secretName string, cert certificate.Certificater, kubeClient kubernetes.Interface) (certificate.Certificater, error) {
```

Reply:
well, to be 100% accurate probably should be `CreateOrGetCertificateFromSecret`, too large though?

Reply:
`CreateOrGetCertificateFromSecret` looks good